From 3e6d2d3de089f5b2427a845f229cdb8278231954 Mon Sep 17 00:00:00 2001
From: Ruchi Kandoi <kandoiruchi@google.com>
Date: Tue, 6 Nov 2018 16:16:17 -0800
Subject: [PATCH 05/10] Prevent Out of bound error in
 llcp_dlc_proc_rr_rnr_pdu()

Test: Nfc Enable/disable
Bug: 116788646
Change-Id: Iefa9ca0a291e0c59a442befccfaa2f17eecd461e
(cherry picked from commit 59c3090aedd725f10e413fe7b38aff67a1782f1d)
---
 src/nfc/llcp/llcp_dlc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nfc/llcp/llcp_dlc.c b/src/nfc/llcp/llcp_dlc.c
index d2a1619..e21b8df 100644
--- a/src/nfc/llcp/llcp_dlc.c
+++ b/src/nfc/llcp/llcp_dlc.c
@@ -1098,7 +1098,11 @@ static void llcp_dlc_proc_rr_rnr_pdu (UINT8 dsap, UINT8 ptype, UINT8 ssap, UINT1
     {
         error_flags = 0;
 
-        rcv_seq = LLCP_GET_NR (*p_data);
+        if (length == 0) {
+            android_errorWriteLog(0x534e4554, "116788646");
+            return;
+        }
+        rcv_seq = LLCP_GET_NR(*p_data);
 
         if (length != LLCP_PDU_RR_SIZE - LLCP_PDU_HEADER_SIZE)
         {
-- 
2.7.4

